src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfCie.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Internal.TypeSystem;
+using static ILCompiler.ObjectWriter.DwarfNative;
+
+namespace ILCompiler.ObjectWriter
+{
+    internal sealed class DwarfCie
+    {
+        public byte PointerEncoding;
+        public byte LsdaEncoding;
+        public byte PersonalityEncoding;
+        public string PersonalitySymbolName;
+        public uint CodeAlignFactor;
+        public int DataAlignFactor;
+        public bool IsSignalFrame;
+        public bool FdesHaveAugmentationData;
+        public byte ReturnAddressRegister;
+        public byte[] Instructions;
+        public byte InitialCFAOffset;
+
+        public DwarfCie(TargetArchitecture targetArchitecture)
+        {
+            IsSignalFrame = false;
+
+            // Each FDE has LSDA pointer
+            FdesHaveAugmentationData = true;
+
+            // Unused
+            PersonalityEncoding = 0;
+            PersonalitySymbolName = null;
+
+            // NOTE: Apple linker only knows how to handle DW_EH_PE_pcrel in combination with
+            // DW_EH_PE_sdata4 or DW_EH_PE_ptr.
+            PointerEncoding = DW_EH_PE_pcrel | DW_EH_PE_sdata4;
+            LsdaEncoding = DW_EH_PE_pcrel | DW_EH_PE_sdata4;
+
+            switch (targetArchitecture)
+            {
+                case TargetArchitecture.ARM64:
+                    CodeAlignFactor = 1;
+                    DataAlignFactor = -4;
+                    ReturnAddressRegister = 30; // LR
+                    Instructions = new byte[]
+                    {
+                        DW_CFA_def_cfa,
+                        31, // SP
+                        0, // Offset from SP
+                    };
+                    InitialCFAOffset = 0;
+                    break;
+
+                //case TargetArchitecture.ARM:
```

Comment:
nit: Maybe just add a "TODO" comment on implementing the 32-bit archs and the reason why it is left out in the current state.

Reply:
I think throwing NYI would be sufficient to remind that 32bit does not work.

If there are plans to actually implement 32bit support, there should be an issue logged for this. `TODOs` are not the best way to track remaining work as context can be quickly lost and then TODOs tend to stay in the code forever.

Reply:
Fair, an exception would work better.

Reply:
I already throw exceptions for that in other places, so exception would be consistent with the rest of the code.

> If there are plans to actually implement 32bit support, there should be an issue logged for this.

I'm not aware of any such plans, the NativeAOT 32-bit code was community supported and bit rotten for years. I made some minimal effort to support some 32-bit platforms. For example win-x86 part of ObjWriter actually works and I tested it enough to produce "Hello World" executable (exception handling, unwinding, etc. is still likely broken though, so this is not endorsement of win-x86 support as a whole). For Linux/ELF the code is structured in a way to allow writing 32-bit ELF files but the architecture specific bits are missing (apparently, CoreCLR doesn't support linux-x86, which only leaves linux-arm32 as potential target, and that's incredibly difficult one to build even for limited testing on contemporary x64/arm64 HW).

Reply:
For posterity, I added basic ELF/ARM32 support. It will run into this exception if you try to use it.

I still don't intend to resurrect the full 32-bit support on Linux or Windows but I also don't want to make it harder if anyone ever tries. It uncovered one mistake in entry size calculation for ELF32.

I submitted all the other 32-bit bit-rot cleanups in a separate PR.